src/java.desktop/macosx/classes/com/apple/laf/AquaFileChooserUI.java

```diff
@@ -2057,7 +2056,6 @@ String getApproveButtonToolTipText(final JFileChooser fc) {
         String getApproveButtonToolTipText(final JFileChooser fc, final String fallbackText) {
             final String tooltipText = fc.getApproveButtonToolTipText();
             if (tooltipText != null) {
-                tooltipText.trim();
                 if (!tooltipText.isEmpty()) return tooltipText;
```

Comment:
In other L&F like Metal, Basic even empty text is allowed.. there's no check for isEmpty()....Should we do the same here too?

Reply:
Yes, I think it would be better to allow to set empty text to conform with other LaF behavior.

Also `JFileChooser.getApproveButtonText()` and `JFileChooser.getApproveButtonToolTipText()` will now return correct values for empty strings.
e.g. before the fix if we call `setApproveButtonText("")` subsequent call to `getApproveButtonText()` will return passed string, but in fact fallback string is used.

I've also labeled the issue with `noreg-hard`.

Reply:
Is it possible to set an empty label in the native file chooser on macOS?

Reply:
Yes, and it looks the same as the one with `" "` label.
<img width="662" alt="image" src="https://user-images.githubusercontent.com/77687766/112301125-ea140e80-8c99-11eb-8588-b1730d43124f.png">